bluesky/run_engine.py

```diff
@@ -486,11 +488,9 @@ def __call__(self, plan, subs=None, **metadata):
                                      "{0}".format(func))
                 self._temp_callback_ids.add(self.subscribe(name, func))
 
-        metadata['scan_type'] = getattr(type(plan), '__name__')
-        if hasattr(plan, 'md'):
-            self._metadata_per_call.update(plan.md)
-        # If kwargs to __call__ collide with plan.md, kwargs win.
-        self._metadata_per_call.update(metadata)
+        metadata_kw['scan_type'] = getattr(type(plan), '__name__')
+        self._plan = plan
+        self._metadata_per_call = metadata_kw
```

Comment:
We used `.update()` instead of `=` so that any references to this object are not blown away. It is guaranteed to be an empty dict because we call `self._clear_call_cache()` above.